### PR TITLE
CI caching android

### PR DESCRIPTION
**Pull request by markholland**
Caching Android and Ruby dependencies in CI

### CODE REVIEW

**Comment by markholland** [comment]
Pointing this at the fix android local branch as didn't want to experiment directly there but this should probably go with that to master.

**Comment by markholland** [comment]
3m30s => 2m45s for the android-alpha job

File: .circleci/config.yml [hunk 1]
```diff
@@ -45,16 +74,23 @@ jobs:
     working_directory: ~/project/android
     docker:
       - image: circleci/android:api-27-node8-alpha
+    environment:
+      TERM: dumb
```
Comment (frigus02) [inline .circleci/config.yml]: What does `TERM: dumb` do?

Comment (markholland) [reply .circleci/config.yml]: This CircleCI image doesn't set anything for TERM and gradle needs it to be set. I assume Fastlane sets it aswell but now that we're calling gradle directly earlier in the job this was needed. The value dumb is abitrary but recommended.

**Pull request merged**